src/component/Tooltip.tsx

```diff
@@ -90,82 +90,78 @@ export type TooltipProps<TValue extends ValueType, TName extends NameType> = Def
-    animationEasing: 'ease',
-    animationDuration: 400,
-    filterNull: true,
-    useTranslate3d: false,
-  };
+const tooltipDefaultProps = {
+  active: false,
+  allowEscapeViewBox: { x: false, y: false },
+  reverseDirection: { x: false, y: false },
+  offset: 10,
+  viewBox: { x: 100, y: 100, height: 50, width: 50 },
+  coordinate: { x: 0, y: 0 },
+  wrapperStyle: {},
+  // this doesn't exist on TooltipProps
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  cursorStyle: {},
+  separator: ' : ',
+  contentStyle: {},
+  itemStyle: {},
+  labelStyle: {},
+  isAnimationActive: !Global.isSsr,
+  animationEasing: 'ease',
+  animationDuration: 400,
+  filterNull: true,
+  useTranslate3d: false,
+  cursor: true,
+};
 
-  state = {
+const Tooltip = <TValue extends ValueType, TName extends NameType>(
+  props: TooltipProps<TValue, TName> & { children?: React.ReactNode },
+) => {
+  const [state, setState] = useState({
     boxWidth: -1,
     boxHeight: -1,
     dismissed: false,
     dismissedAtCoordinate: { x: 0, y: 0 },
-  };
+  });
 
-  private wrapperNode: HTMLDivElement;
-
-  componentDidMount() {
-    this.updateBBox();
-  }
-
-  componentDidUpdate() {
-    this.updateBBox();
-  }
+  const wrapperNode = useRef<HTMLDivElement>();
+  const { allowEscapeViewBox, reverseDirection, coordinate, offset, position, viewBox } = props;
 
-  updateBBox() {
-    const { boxWidth, boxHeight, dismissed } = this.state;
+  const updateBBox = () => {
+    const { boxWidth, boxHeight, dismissed } = state;
     if (dismissed) {
-      this.wrapperNode.blur();
-      if (
-        this.props.coordinate.x !== this.state.dismissedAtCoordinate.x ||
-        this.props.coordinate.y !== this.state.dismissedAtCoordinate.y
-      ) {
-        this.setState({ dismissed: false });
+      wrapperNode.current.blur();
+      if (coordinate.x !== state.dismissedAtCoordinate.x || coordinate.y !== state.dismissedAtCoordinate.y) {
+        setState(prev => ({ ...prev, dismissed: false }));
       }
     } else {
-      this.wrapperNode.focus({ preventScroll: true });
+      wrapperNode.current?.focus({ preventScroll: true });
     }
 
-    if (this.wrapperNode && this.wrapperNode.getBoundingClientRect) {
-      const box = this.wrapperNode.getBoundingClientRect();
+    if (wrapperNode.current && wrapperNode.current.getBoundingClientRect) {
+      const box = wrapperNode.current.getBoundingClientRect();
 
       if (Math.abs(box.width - boxWidth) > EPS || Math.abs(box.height - boxHeight) > EPS) {
-        this.setState({
+        setState(prev => ({
+          ...prev,
           boxWidth: box.width,
           boxHeight: box.height,
-        });
+        }));
       }
     } else if (boxWidth !== -1 || boxHeight !== -1) {
-      this.setState({
+      setState(prev => ({
+        ...prev,
         boxWidth: -1,
         boxHeight: -1,
-      });
+      }));
     }
-  }
+  };
+
+  useEffect(() => {
+    updateBBox();
+  }, []);
```

Comment:
Previously `updateBBox()` was called both on onComponentMount as well as onComponentUpdate.

This is not equivalent. The useEffect will only fire onMount, but not onUpdate. 
If the behaviour should replicate, then state would have to be added into the dependency array.